consumer/src/main/java/com/flipkart/varadhi/consumer/GroupedMessageSrc.java

```diff
@@ -158,7 +214,7 @@ private void free(String groupId, MessageConsumptionStatus status) {
                     throw new IllegalStateException(String.format("Tried to free group %s: %s", gId, tracker));
                 }
                 var messages = tracker.messages;
-                if (!messages.isEmpty() && messages.getFirst().remaining() == 0) {
```

Comment:
using `while` to make sure that messages batch with empty entries are not going to be put back in the hashmap.